ssc/cmod_merchantplant_eqns.cpp

```diff
@@ -41,32 +42,32 @@ void mp_ancillary_services(ssc_data_t data)
 		{ SSC_INPUT,        SSC_NUMBER,     "mp_enable_ancserv4",		      "Enable ancillary services 4 revenue",   "0/1",   "",    "",  "*",	"INTEGER,MIN=0,MAX=1",      "" },
 		{ SSC_INPUT, SSC_MATRIX, "mp_ancserv4_revenue", "Ancillary services 4 revenue input", "", "","*", "", "" },
 		*/
-		VT_GET_INPUT(vt, "analysis_period", analysis_period)
-		VT_GET_INPUT(vt, "mp_enable_energy_market_revenue", mp_enable_energy_market_revenue)
-		VT_GET_INPUT(vt, "mp_enable_ancserv1", mp_enable_ancserv1)
-		VT_GET_INPUT(vt, "mp_enable_ancserv2", mp_enable_ancserv2)
-		VT_GET_INPUT(vt, "mp_enable_ancserv3", mp_enable_ancserv3)
-		VT_GET_INPUT(vt, "mp_enable_ancserv4", mp_enable_ancserv4)
-		VT_GET_INPUT(vt, "mp_energy_market_revenue", mp_energy_market_revenue)
-		VT_GET_INPUT(vt, "mp_ancserv1_revenue", mp_ancserv1_revenue)
-		VT_GET_INPUT(vt, "mp_ancserv2_revenue", mp_ancserv2_revenue)
-		VT_GET_INPUT(vt, "mp_ancserv3_revenue", mp_ancserv3_revenue)
-		VT_GET_INPUT(vt, "mp_ancserv4_revenue", mp_ancserv4_revenue)
-		gen_is_assigned = (vt->lookup("gen") != NULL);
+		vt_get_double(vt, "analysis_period", &analysis_period);
+        vt_get_int(vt, "mp_enable_energy_market_revenue", &mp_enable_energy_market_revenue);
+        vt_get_int(vt, "mp_enable_ancserv1", &mp_enable_ancserv1);
+        vt_get_int(vt, "mp_enable_ancserv2", &mp_enable_ancserv2);
+        vt_get_int(vt, "mp_enable_ancserv3", &mp_enable_ancserv3);
+        vt_get_int(vt, "mp_enable_ancserv4", &mp_enable_ancserv4);
+        vt_get_matrix(vt, "mp_energy_market_revenue", mp_energy_market_revenue);
+        vt_get_matrix(vt, "mp_ancserv1_revenue", mp_ancserv1_revenue);
+        vt_get_matrix(vt, "mp_ancserv2_revenue", mp_ancserv2_revenue);
+        vt_get_matrix(vt, "mp_ancserv3_revenue", mp_ancserv3_revenue);
+        vt_get_matrix(vt, "mp_ancserv4_revenue", mp_ancserv4_revenue);
+        gen_is_assigned = (vt->lookup("gen") != NULL);
 		if (gen_is_assigned)
 		{
 			system_capacity = 0.0;
-			VT_GET_INPUT(vt, "gen", system_gen)
+            vt_get_matrix(vt, "gen", system_gen);
 		}
 		else
 		{
-			VT_GET_INPUT(vt, "system_capacity", system_capacity)
+			vt_get_double(vt, "system_capacity", &system_capacity);
 		}
 		calculate_revenue = (vt->lookup("mp_calculate_revenue") != NULL);
 		if (calculate_revenue)
 		{
-			VT_GET_INPUT(vt, "mp_calculate_revenue", mp_calculate_revenue)
-				calculate_revenue = (mp_calculate_revenue > 0.5);
+			vt_get_double(vt, "mp_calculate_revenue", &mp_calculate_revenue);
+            calculate_revenue = (mp_calculate_revenue > 0.5);
```

Comment:
Should be an int or boolean - check latest develop if we are going to remove the VT_GET_INPUT macro and do explicit type casting.

Reply:
I don't see "mp_calculate_revenue" in the [docs ](https://github.com/NREL/ssc/blob/develop/ssc/cmod_merchantplant_eqns.h).

I changed the docs to add `gen` and `mp_calculate_revenue` and use `vt_get_int` in 281a7c3ddde47e448dd5eb1411b3290e8b388d4c.

@janinefreeman Could you add some description to what each input in `mp_ancillary_services_doc` means?

Reply:
As a side note, when you're using an equation from within a compute_module as in [cmod_merchantplant](https://github.com/NREL/ssc/blob/c7c0e56a913c80e7737525654deb0a86d29a3696/ssc/cmod_merchantplant.cpp#L1023), you can pass in the cmod's vartable directly. Instead of looking up each input to the eqn and inserting it into a new vartable, you can just pass it the original vartable after inserting whichever eqn parameters are missing. so in the above example, you can replace line 1023-1040 with 
```
assign("mp_calculate_revenue", var_data(ssc_number_t(1.0)));
mp_ancillary_services(m_vartab);
if (lookup("mp_ancillary_services") ...
```